### ISSUE TITLE
Version 3.46 broken repository checkout

### ISSUE THREAD

**Issue by hfp** [buildkite/agent#2073]
With version 3.46 there are failing checkouts exhausting all retries like `Warning: Checkout failed! exit status 128 (Attempt 1/3 Retrying in 2s)`. Before the warning, the following output appears:

```text
$ git clean -ffxdq
# Checking to see if Git data needs to be sent to Buildkite
$ buildkite-agent meta-data exists buildkite:git:commit
# Sending Git commit information back to Buildkite
```

It seems `Sending Git commit information back to Buildkite` is causing an exit status of `128`, which in turn causes to enter a retry-loop.

**Comment by moskyb** [comment]
hey @hfp! we've spent a little time looking into this (thanks for sending me the link btw), and it appears to be an issue with the version of git that you're using (which appears to be CentOS 7's security-updated version of Git v1.8.3) doesn't support the use of the `--no-patch` flag we added in #2054. `--no-patch` is just a synonym for `-s` though, so we're gonna change it back to using `-s`.

we don't provide a compatibility promise for git, so we have to do our best to support legacy versions of git for use with the agent, but most of our testing uses more modern versions, so we're biased towards using those in the agent. this means that sometimes things might break on older versions of git, but we'll do our best to address these as they come up.
